spec/realtime/channel.test.js

```diff
@@ -1506,5 +1506,35 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
 				}
 			);
 		});
+
+                // RTL5j
+                it('detaching from suspended channel transitions channel to detached state', function (done) {
+                    var realtime = helper.AblyRealtime({ transports: [helper.bestTransport] });
+                    var channelName = 'detach_from_suspended';
+                    var channel = realtime.channels.get(channelName);
+
+                    channel.state = 'suspended';
```

Comment:
Just to ensure my understanding: I'm assuming this isn't public API - i.e. the test is mutating an internal object?

Reply:
It certainly is part of the public API and is documented in the typescript interface definition, but I think it should be clear to developers that the intention is for this to be readonly and I would expect no one would actually try mutating it unless they're trying to mess with the lib internals.

Reply:
typescript has a `readonly` attribute modifier now, we can add that to the typings on things like this to make the intention even clearer